### PR TITLE
Feature/data download

### PR DESCRIPTION
**Pull request by antimaLinux**
## ADD
- added download feature

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# E222 multiple spaces after operator
+# E241 multiple spaces after ':'
+# E251 unexpected spaces around keyword / parameter equals
+# E272 multiple spaces before keyword
+# E722 do not use bare except
+# E731 do not assign a lambda expression, use a def
+# W503 line break occurred before a binary operator (badly implemented, doesn't allow log operators within brackets)
+
+
+ignore = E114,E116,E203,E221,E222,E241,E251,E272,E722,E731,W503
```
Comment (drugilsberg) [inline .flake8]: maybe add newline

File: kickeststats/helpers/data.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from itertools import zip_longest
+from typing import Iterable, Any
+
+
+def grouper(iterable: Iterable, n: int, fillvalue: Any = None) -> Iterable:
```
Comment (drugilsberg) [inline kickeststats/helpers/data.py:5]: classic grouper :)

**Review by drugilsberg — COMMENTED** [review]
Will pull and change the single to double quotes also in teh remaining files.

File: kickeststats/constants.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+from kickeststats.exceptions import EnvVariableNotSet
+
+CHROMEDRIVER_EXECUTABLE_PATH = os.environ.get(
+    "CHROMEDRIVER_EXECUTABLE_PATH",
+    pkg_resources.resource_filename(
+        "kickeststats", "resources/drivers/chromedriver"
+    )
+)
+
+if not CHROMEDRIVER_EXECUTABLE_PATH:
```
Comment (drugilsberg) [inline kickeststats/constants.py:13]: not needed anymore, but we can keep it

File: kickeststats/download.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
+from splinter import Browser  # type: ignore
+from splinter.driver.webdriver.chrome import WebDriver as ChromeWebDriver  # type: ignore
+from splinter.driver.webdriver import WebDriverElement  # type: ignore
+from typing import List
+import time
+
+from .helpers.parsers import HeaderParser, RowParser, PaginationParser
```
Comment (drugilsberg) [inline kickeststats/download.py:7]: changed to relative imports

File: setup.py [hunk 5]
```diff
@@ -5,20 +5,25 @@
...
-            'py.typed',
-            'resources/tests/players.jsonl'
+        "kickeststats": [
+            "py.typed",
+            "resources/tests/players.jsonl",
+            "resources/drivers/chromedriver"
         ]
     },
-    install_requires=['pandas==1.1.2']
+    install_requires=[
```
Comment (drugilsberg) [inline setup.py]: added dependencies here for installation from PYPI.

**Review by drugilsberg — APPROVED** [review]

File: kickeststats/helpers/data.py (hunk 2 above)
Comment (antimaLinux) [reply kickeststats/helpers/data.py:5]: yes! itertools recipe! why don't they put the recipes in the package? why? :(

**Pull request merged**